src/powerbox/dft.py

```diff
@@ -25,45 +25,52 @@
 one must be careful that the frequencies returned are descending, rather than ascending).
 """
 
-import warnings
-
 __all__ = ["fft", "ifft", "fftfreq", "fftshift", "ifftshift"]
 
-# Try importing the pyFFTW interface
-try:
-    from multiprocessing import cpu_count
+# To avoid MKL-related bugs, numpy needs to be imported after pyfftw: see https://github.com/pyFFTW/pyFFTW/issues/40
+import numpy as np
+import warnings
 
-    THREADS = cpu_count()
+from .dft_backend import get_fft_backend
 
-    from pyfftw.interfaces.cache import enable, set_keepalive_time
-    from pyfftw.interfaces.numpy_fft import fftfreq as _fftfreq
-    from pyfftw.interfaces.numpy_fft import fftn as _fftn
-    from pyfftw.interfaces.numpy_fft import fftshift as _fftshift
-    from pyfftw.interfaces.numpy_fft import ifftn as _ifftn
-    from pyfftw.interfaces.numpy_fft import ifftshift as _ifftshift
 
-    def fftn(*args, **kwargs):
-        return _fftn(*args, threads=THREADS, **kwargs)
+def fftshift(x, *args, **kwargs):
+    try:
+        nthreads = kwargs["nthreads"]
+    except KeyError:
+        nthreads = None
+    backend = get_fft_backend(nthreads)
```

Comment:
This can actually be reduced to `backend = get_fft_backend(kwargs.get('nthreads'))`

Reply:
Since the `.get()` returns None if the key doesn't exist